### PR TITLE
Add ONNX support for DETR

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

This PR adds ONNX support for DETR and for object-detection models in general. Linked to #16308 and discussed [here](https://huggingface.co/facebook/detr-resnet-50/discussions/1).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/onnx/features.py
Comment (lewtun) [inline src/transformers/onnx/features.py:234]: Does it make sense to include support for the segmentation head as well?

**Review by lewtun — COMMENTED** [review]
Thanks for adding this popular model @regisss !

I left a small question, but otherwise this looks great to me. Did you also check the slow tests pass?

File: src/transformers/onnx/features.py
Comment (regisss) [reply src/transformers/onnx/features.py:236]: It absolutely does, thanks for reminding me! My last commit added it. I had to set the ONNX opset at 12 to make it work.

**Comment by regisss** [comment]
All slow tests passed @lewtun

**Comment by regisss** [comment]
Pinging @sgugger for approval

**Review by lewtun — APPROVED** [review]
Thanks for iterating @regisss - LGTM from my side!

Let's wait for Sylvain's blessing before merging :)

**Review by sgugger — APPROVED** [review]
LGTM, thanks for adding this!

**Pull request merged**
